src/components/Map/Map.tsx

```diff
@@ -517,6 +517,13 @@ export default function Map(props: MapProps): JSX.Element {
     return source.entities?.some((entity) => entity?.boundary?.length);
   });
 
+  const renderedPopup = useMemo<JSX.Element | null>(() => {
+    if (!popupInfo || !popupRenderer) {
```

Comment:
I don't really get when this will be undefined if you are validating that they exist before calling this function

Reply:
This is a `useMemo` it is automatically called when the properties change, the properties can be undefined.

Reply:
So the validation in the JSX `{ popupInfo && popupRenderer && ... }` would not be necessary?

Reply:
Those properties are dereferenced in the `Popup` component and need to be validated. example, see lines 568 and 578. Typescript will not assume the properties are valid even if we validated them in some other useMemo component.

Reply:
Got it! Thanks!